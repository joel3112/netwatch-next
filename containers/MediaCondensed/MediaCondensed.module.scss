/** Container (main component) **/

.card {
  width: auto !important;

  &.condensed::before {
    @include box(100%, 100%);
    @include absolute(top 0 left 0);

    content: '';
    display: block;
    z-index: 5;
    background-image: linear-gradient(60deg, rgb(20 20 20 / 85%), transparent 1000px);

    .xs & {
      background-image: linear-gradient(60deg, rgb(20 20 20 / 85%), transparent 300px);
    }
  }
}

.image {
  .condensed & {
    &::before {
      display: none;
    }
  }
}

/** MediaCondensedInfo (child component) **/

.info {
  @include absolute(top 30% left 3em);
  @include flex($direction: column, $gap: 1rem);

  color: var(--color-button-text);
  z-index: 5;
  max-width: 50%;

  .xs & {
    top: auto;
    bottom: 30px;
    left: 2em;
  }
}

.title {
  font-size: rem(25px) !important;
  font-weight: 600 !important;

  .xs &,
  .sm & {
    @include max-lines(2);

    margin-top: -15px;
    font-size: rem(18px) !important;
  }
}

.description {
  .xs &,
  .sm &,
  .md & {
    display: none !important;
  }
}

.actionsInfo {
  display: flex !important;
  justify-content: flex-start !important;
  align-items: center !important;
  position: relative;
  bottom: auto;
  padding: 0 !important;
}

.buttonInfo {
  background-color: var(--color-disabled) !important;
}
