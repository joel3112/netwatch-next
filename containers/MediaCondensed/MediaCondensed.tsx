import { ElementHTML, MediaData, MediaImageRatio } from '@/types';
import Link from 'next/link';
import { RiPlayFill } from 'react-icons/ri';
import { useI18n } from '@/hooks/useI18n';
import { useBreakpoint } from '@/hooks/useBreakpoint';
import { useModal } from '@/hooks/useModal';
import { useVideo } from '@/hooks/useVideo';
import { useFavourite } from '@/hooks/useFavourite';
import { Space } from '@/components/layout';
import { Text } from '@/components/typography';
import { Card } from '@/components/display';
import { Button } from '@/components/forms';
import { Modal, Portal } from '@/components/overlay';
import { Video } from '@/components/media';
import { classes } from '@/utils/helpers';
import styles from '@/containers/MediaCondensed/MediaCondensed.module.scss';

/* -------------------------------------------------------------------------- */
/** MediaCondensedInfo (main component) **/
/* -------------------------------------------------------------------------- */

type MediaCondensedInfoProps = MediaData;

const MediaCondensedInfo = (props: MediaCondensedInfoProps) => {
  const { id, type, name, description } = props;
  const { t } = useI18n('home');
  const { isOpened, handleChange } = useModal();
  const { videoId } = useVideo(isOpened ? id : 0, type);
  const { favouriteAction, FavouriteIcon, onToggle } = useFavourite();

  return (
    <>
      <div className={styles.info} data-swiper-parallax="-300">
        <Text className={styles.title}>{name}</Text>

        <Text size="sm" maxLines={4} className={styles.description}>
          {description}
        </Text>

        <Space align="center" gap={8}>
          <Card.Actions className={styles.actionsInfo}>
            <Card.Actions.Item
              icon={FavouriteIcon(id)}
              tooltip={favouriteAction(id)}
              onClick={(e: UIEvent) => onToggle(e, props)}
            />
            <Card.Actions.Item
              tooltip={t('item.action.trailer')}
              icon={RiPlayFill}
              onClick={() => handleChange(true)}
            />
          </Card.Actions>

          <Link href={{ pathname: '/[type]/[id]', query: { type, id } }}>
            <a>
              <Button rounded className={styles.buttonInfo}>
                {t('item.action.more')}
              </Button>
            </a>
          </Link>
        </Space>
      </div>

      <Portal>
        <Portal.Paper>
          <Modal opened={isOpened} onChange={handleChange}>
            <Video id={videoId} autoplay />
          </Modal>
        </Portal.Paper>
      </Portal>
    </>
  );
};

/* -------------------------------------------------------------------------- */
/** Container (main component) **/
/* -------------------------------------------------------------------------- */

export type MediaCondensedProps = typeof defaultProps & ElementHTML & MediaData;

const defaultProps = {};

const MediaCondensed = (props: MediaCondensedProps) => {
  const { key, isMobile } = useBreakpoint();

  return (
    <div className={classes(styles.wrapper, styles[key])}>
      <Card className={classes(styles.card, styles.condensed)} skeleton={!props.type}>
        <Card.Image
          classes={styles.image}
          src={props.backdrop}
          width="100%"
          ratio={isMobile ? MediaImageRatio.BACKDROP : MediaImageRatio.BACKDROP * 0.6}
          lazy
        />
      </Card>

      <MediaCondensedInfo {...props} />
    </div>
  );
};

MediaCondensed.defaultProps = defaultProps;

export default MediaCondensed;
