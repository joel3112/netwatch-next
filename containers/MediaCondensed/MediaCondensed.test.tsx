import { render } from '@testing-library/react';
import MediaCondensed from '@/containers/MediaCondensed/MediaCondensed';

describe('Tests MediaCondensed component', () => {
  test('renders component correctly', () => {
    const { container } = render(
      <MediaCondensed
        id={1}
        type="movie"
        name="Pelicula"
        image="https://picsum.photos/200/300"
        route="movie"
      />
    );

    expect(container).toMatchSnapshot();
  });
});
