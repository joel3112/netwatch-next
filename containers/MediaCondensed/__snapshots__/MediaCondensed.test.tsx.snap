// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests MediaCondensed component renders component correctly 1`] = `
<div>
  <div
    class="wrapper md"
  >
    <div
      class="wrapper card condensed"
    >
      <div
        aria-label="image"
        class="wrapper"
        role="img"
      >
        <div
          class="image content"
        />
        <style>
          
        .wrapper {
          width: 100%;
          height: 100%;
          min-width: 100%;
        }
        .wrapper::before {
          padding-top: 33.6%;
        }
      
        </style>
      </div>
    </div>
    <div
      class="info"
      data-swiper-parallax="-300"
    >
      <span
        class="wrapper title md"
      >
        Pelicula
        <style>
          
        .wrapper {
          
          
          
        }
      
        </style>
      </span>
      <span
        class="wrapper description sm maxLines"
      >
        <style>
          
        .wrapper {
          
          
          -webkit-line-clamp: 4;
        }
      
        </style>
      </span>
      <div
        class="wrapper"
      >
        <div
          class="wrapper actions actionsInfo"
        >
          <div
            aria-label="favourites.add"
            class="content"
            data-mui-internal-clone-element="true"
          >
            <button
              aria-label="action"
              class="wrapper medium clear action"
            >
              <svg
                class="actionIcon"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M416 277.333H277.333V416h-42.666V277.333H96v-42.666h138.667V96h42.666v138.667H416v42.666z"
                />
              </svg>
            </button>
          </div>
          <div
            aria-label="item.action.trailer"
            class="content"
            data-mui-internal-clone-element="true"
          >
            <button
              aria-label="action"
              class="wrapper medium clear action"
            >
              <svg
                class="actionIcon"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g>
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M19.376 12.416L8.777 19.482A.5.5 0 0 1 8 19.066V4.934a.5.5 0 0 1 .777-.416l10.599 7.066a.5.5 0 0 1 0 .832z"
                  />
                </g>
              </svg>
            </button>
          </div>
          <style>
            
        .wrapper {
          display: flex;
          flex-direction: row;
          flex-wrap: nowrap;
          gap: 8px;
          justify-content: flex-end;
          
        }
      
          </style>
        </div>
        <a
          href="/movie"
        >
          <button
            class="wrapper medium rounded buttonInfo"
          >
            item.action.more
          </button>
        </a>
        <style>
          
        .wrapper {
          display: flex;
          flex-direction: row;
          flex-wrap: nowrap;
          gap: 8px;
          
          align-items: center;
        }
      
        </style>
      </div>
    </div>
    <div
      class="wrapper"
    />
  </div>
</div>
`;
